webapp/public/javascripts/controllers.js

```diff
@@ -1,15 +1,6 @@
 function MainPageController($scope, $http, $location){
-    $scope.signUp = function(){
-        $http.post('/api/users/register', {email: $scope.email, password: $scope.password}).
-            success(function(data, status, headers, config) {
-
-                $location.path('/users')
-            }).
-            error(function(data, status, headers, config) {
-                $scope.errorsMessages = data;
-                $scope.email = "";
-                $scope.password = "";
-            });
+    $scope.createAccount = function(){
+        $location.path('/register', {email: $scope.email, password: $scope.password})
```

Comment:
Did you verify that {email: $scope.email ....} does what you intend? ie: is the register form prefilled with this data when the user is navigated to /register?